Content.Server/Implants/ImplanterSystem.cs

```diff
@@ -103,17 +106,36 @@ public void TryImplant(ImplanterComponent component, EntityUid user, EntityUid t
     //TODO: Remove when surgery is in
     public void TryDraw(ImplanterComponent component, EntityUid user, EntityUid target, EntityUid implanter)
     {
-        var args = new DoAfterArgs(EntityManager, user, component.DrawTime, new DrawEvent(), implanter, target: target, used: implanter)
-        {
-            BreakOnUserMove = true,
-            BreakOnTargetMove = true,
-            BreakOnDamage = true,
-            NeedHand = true,
-        };
-
-        if (_doAfter.TryStartDoAfter(args))
+        // Checks for sleeping but let slime people be awake. TODO: generalize this to use the organType when metabolism is reworked / when consciousness states are added.
+        if (TryComp<HumanoidAppearanceComponent>(target, out var appearance) && appearance.Species != "SlimePerson"){
+            var args = new DoAfterArgs(EntityManager, user, component.DrawTime, new DrawEvent(), implanter, target: target, used: implanter)
+            {
+                BreakOnUserMove = true,
+                BreakOnTargetMove = true,
+                BreakOnDamage = true,
+                BreakOnTargetWake = true,
+                NeedHand = true,
+            };
+            
+            if (_doAfter.TryStartDoAfter(args) && HasComp<SleepingComponent>(target)){
             _popup.PopupEntity(Loc.GetString("injector-component-injecting-user"), target, user);
-
+            } else {
+            _popup.PopupEntity(Loc.GetString("injector-component-implant-draw-fail-not-asleep"), target, user);
+            }
+            
+        } else {
+            var args = new DoAfterArgs(EntityManager, user, component.DrawTime, new DrawEvent(), implanter, target: target, used: implanter)
+            {
+                BreakOnUserMove = true,
+                BreakOnTargetMove = true,
+                BreakOnDamage = true,
+			    BreakOnTargetWake = false,
+                NeedHand = true,
+            };
+            
+            if (_doAfter.TryStartDoAfter(args))
+            _popup.PopupEntity(Loc.GetString("injector-component-injecting-user"), target, user);
+        }
     }
```

Comment:
1. Don't mix tabs and spaces.
2. Don't hardcode this for species, use a component if yo ureally want it.
3. Please don't inline your ifs like this it makes the formatting a mess.